packages/view-types/statistical-plots/src/ExpressionHistogramSubscriber.js

```diff
@@ -81,12 +100,34 @@ export function ExpressionHistogramSubscriber(props) {
         const values = obsFeatureMatrix.data
           .subarray(cellIndex * numGenes, (cellIndex + 1) * numGenes);
         const sumValue = sum(values) * 100 / 255;
-        return { value: sumValue, gene: null };
+        const newItem = { value: sumValue, gene: null, cellId };
+        dataOnSelect.push(newItem);
+        setDataOnSelect(dataOnSelect);
+        return newItem;
       });
     }
     return null;
   }, [obsIndex, featureIndex, obsFeatureMatrix, firstGeneSelected, expressionData]);
 
+
+  const onSelect = (value) => {
+    const getCellIdsInRange = (range) => {
+      const [lowerBound, upperBound] = range;
+
+      return dataOnSelect
+        .filter(item => item.value >= lowerBound && item.value <= upperBound)
+        .map(item => item.cellId);
+    };
+
+    const selectedCellIds = getCellIdsInRange(value);
+    setObsSelection(
+      selectedCellIds, additionalCellSetsRef.current, cellSetColor,
+      setCellSetSelection, setAdditionalCellSets, setCellSetColor,
+      setCellColorEncoding,
+      'Selection based on transcript count',
+    );
+  };
```

Comment:
```suggestion
  const onSelect = useCallback((value) => {
    const getCellIdsInRange = (range) => {
      const [lowerBound, upperBound] = range;

      return data
        .filter(item => item.value >= lowerBound && item.value <= upperBound)
        .map(item => item.cellId);
    };

    const selectedCellIds = getCellIdsInRange(value);
    setObsSelection(
      selectedCellIds, additionalCellSets, cellSetColor,
      setCellSetSelection, setAdditionalCellSets, setCellSetColor,
      setCellColorEncoding,
      'Selection based on transcript count',
    );
  }, [additionalCellSets, cellSetColor, data, setAdditionalCellSets,
    setCellColorEncoding, setCellSetColor, setCellSetSelection,
  ]);
```

Paired with the other suggestion to pass `latestOnSelect` as a parameter to the debounced function, this can become a regular `useCallback` with no need for `dataOnSelect` or `additionalCellSetsRef`.